lib/ddtrace/tracing/runtime.rb

```diff
@@ -0,0 +1,90 @@
+require 'forwardable'
+require 'ddtrace/configuration/settings'
+require 'ddtrace/configuration/components'
+require 'ddtrace/configuration/pin_setup'
+
+module Datadog
+  module Tracing
+    # This class encapsulates all tracer runtime data and objects.
+    #
+    # Any stateful component or information that the tracer requires
+    # to execute must be directly or indirectly managed by {Runtime}.
+    #
+    # Any object who's life cycle is not captured by {Runtime} should be
+    # part of the generic tracer structure, unaffected by configuration
+    # or runtime state.
+    #
+    # Creating a new {Runtime} will effectively configure and initialized a new
+    # running instance of the tracer.
+    #
+    # Reconfiguration the {Runtime} will affect the running tracer instance without
+    # a complete restart of the tracer internals.
+    #
+    # Destroying the {Runtime} will shutdown the current tracer instance.
+    #
+    # The tracer has started if and only if (iff) the {Runtime} has been started.
+    # The tracer is shut down if and only if (iff) the {Runtime} is shut down.
+    #
+    # A production user of the tracer will only initialize at most a single {Runtime}
```

Comment:
will only => must only? (see also https://datatracker.ietf.org/doc/html/rfc2119 )

Reply:
It's interesting because there's no choice here: "A production user will always initialize at most one instance of Runtime". It's a consequence of our design here, by construction.

"A production user must initialize at most one instance of Runtime" seems like a requirement that has to be fulfilled in order for the predicate to be compliant, but there's no way for this requirement to not be met.